### DISCUSSION TITLE
Did anyone manage to use ord on Raspberry Pi?

### DISCUSSION THREAD

**Discussion by DanielHook15** [ordinals/ord#2410] [Q&A]
I have a Raspberry Pi 4. 
With Raspbian 64bit, it was not possible to get Bitcoin Core running. 
With Raspbian 32bit, installation of Bitcoin Core was easy, installation of ord was possible when compiling from source, but indexing does not work (crashes at block 300000 with the problem described in https://github.com/casey/ord/issues/1674).

Did anyone get ord running on a Raspberry Pi? If yes, which OS?

**Comment by bnonni** [comment]
@DanielHook15 that doesn't sound right. From my experience, its the opposite. You wont be able to build Bitcoin Core with a 32 bit OS. 64 bit OS is pretty standard across the board, and I know rpi 3, 4, and Zero all support a 64 bit OS (https://arstechnica.com/gadgets/2022/02/64-bit-raspberry-pi-os-exits-beta-is-available-for-all-pi-3-4-and-zero-2-boards/)

Yes -- I helped a friend get ord running on a raspberry pi with MyNode OS on it. However, you should be able to build from source with just about modern OS, so long as it has rust installed (https://www.rust-lang.org/tools/install). My friend and I had to build from source because the mynode OS failed the install script test, which is looking for 1 of these 3 cpu arch + OS options:
```
    arm64-Darwin) target=aarch64-apple-darwin;;
    x86_64-Darwin) target=x86_64-apple-darwin;;
    x86_64-Linux) target=x86_64-unknown-linux-gnu;;
```

To build from source (https://github.com/casey/ord#building)
```
# these command build ord
git clone https://github.com/casey/ord.git
cd ord
cargo build --release

# this command adds ord to your path
export PATH="`which ord`:$PATH"

# this command tests the install
ord --version
```

**Comment by DanielHook15** [comment]
Very helpful, thank you!

I downloaded Bitcoin Core from https://bitcoincore.org/en/download/ and clicked on "ARM Linux". Since "32bit 64bit" is written under "ARM Linux", I assumed that I will get one package that works in both cases. As I realized a few minutes ago, however, I got only a version for 32bit. If one wants 64bit, one needs to click on the tiny "64bit" below "ARM Linux". 
As a consequence, it did not work on my Raspbian 64bit and I switched to Raspbian 32bit, where it worked. 
My node is running without any problems since then. 
Adding ord to the node, however, did not work. Thank you for confirming that Bitcoin Core does work with 64bit. 

I will try your suggestion next (but instead of export PATH="`which ord`:$PATH" I will write the PATH into .profile to make it permanent). If that does not work I will post it here. Otherwise the problem is solved.

**Comment by KeepYourCoins** [comment]
I managed to get it running, with some [slight modifications](https://github.com/casey/ord/pull/1562/files) on my pi4 running [raspiblitz](https://github.com/rootzoll/raspiblitz). I did have to grab a copy of the ord index from someone else though - which is far from ideal.

**Comment by raphjaph** [comment]
We just released a new version, which will make building the index much faster

**Comment by DanielHook15** [comment]
Great! How to update ord is explained at https://github.com/casey/ord/issues/1816.

**Comment by mitohund** [comment]
> I managed to get it running, with some [slight modifications](https://github.com/casey/ord/pull/1562/files) on my pi4 running [raspiblitz](https://github.com/rootzoll/raspiblitz). I did have to grab a copy of the ord index from someone else though - which is far from ideal.

This sounds super interesting - I'm hoping to get ord to run on my Raspiblitz myself. I've chatted with a few people about this and one of them told me they managed to get ord to run on their RB by simply disabling rpc auth in bitcoind.conf (to make bitcoind write the cookie auth file ord needs to connect to the node). Hopefully I could then use a pre-built version of ord (since I have never compiled anything myself (yet)). However, that person said that when they rsynced a copy of the database (redb?) from their main node (powerful server, 48 GB RAM) to their Raspiblitz, it didn't work. May I ask how you did it?

More generally: does ord work well on your Raspiblitz?
